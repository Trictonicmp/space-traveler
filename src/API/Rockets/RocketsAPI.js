import { ROCKETS_ENDPOINT } from '../APIconfig';

const getRocketsFromAPI = async (errorCallback = null) => {
  let data = [];
  try {
    const response = await fetch(ROCKETS_ENDPOINT, { method: 'GET' });
    data = await response.json();
  } catch (error) {
    if (errorCallback) errorCallback();
  }

  return data;
};

export default getRocketsFromAPI;
