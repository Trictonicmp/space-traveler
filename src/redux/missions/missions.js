// ACTIONS
const FETCH = 'FETCH_MISSION'


// ACTION CREATORS
const FETCH_MISSION = () => async (dispatch) => {
  try {
    
  } catch (error) {
    
  }
}

const initialState = [];

const reducer = (state = initialState, action) => {
  switch (action) {
    case 'FETCH_MISSON':
      return [...state, action.payload];

    default:
      return state;
  }
};

export default reducer;
