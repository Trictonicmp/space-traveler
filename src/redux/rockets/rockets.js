import getRocketsFromAPI from '../../API/Rockets/RocketsAPI';

// ACTIONS
export const GET_ROCKETS = 'space-traveler/rockets/GET_ROCKETS';

// ACTION CREATORS
export const getRockets = () => async (dispatch) => {
  const rockets = await getRocketsFromAPI();

  dispatch({
    type: GET_ROCKETS,
    payload: rockets,
  });
};

const reducer = (state = [], action) => {
  switch (action.type) {
    case GET_ROCKETS:
      return action.payload;
    default:
      return state;
  }
};

export default reducer;
