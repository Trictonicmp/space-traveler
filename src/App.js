import { Routes, Route } from 'react-router-dom';
import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { getRockets } from './redux/rockets/rockets';
import Nav from './components/NavBar/Nav';
import Mission from './components/pages/Mission';
import Rockets from './components/pages/Rockets';
import Profile from './components/pages/Profile';

function App() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getRockets());
  }, []);

  return (
    <>
      <Nav />
      <Routes>
        <Route path="/" element={<Rockets />} />
        <Route path="/mission" element={<Mission />} />
        <Route path="/profile" element={<Profile />} />
      </Routes>
    </>
  );
}

export default App;
