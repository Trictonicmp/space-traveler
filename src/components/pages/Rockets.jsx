import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux/es/exports';
import { getRockets } from '../../redux/rockets/rockets';
import Rocket from '../Rocket/Rocket';

import styles from '../../css/components/RocketsContainer.module.css';

const Rockets = () => {
  const state = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getRockets());
  }, []);

  const rocketComponents = state.map((rocket) => (
    <Rocket
      id={rocket.id}
      key={rocket.id}
      rocketName={rocket.rocketName}
      description={rocket.description}
      imageURL={rocket.imageURL}
    />
  ));
  return (
    <ul className={styles.rocketsContainer}>
      { rocketComponents }
    </ul>
  );
};

export default Rockets;
