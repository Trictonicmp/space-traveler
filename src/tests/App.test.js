import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import rockets from '../redux/rockets/rockets';
import missions from '../redux/missions/missions';
import App from '../App';
import Rockets from '../components/pages/Rockets';

const { configureStore } = require('@reduxjs/toolkit');

const createStore = () => {
  const preloadedState = {
    rockets: [
      {
        id: 1,
        rocketName: 'Falcon Heavy',
        description: 'With the ability to lift into orbit over 54 metric tons (119,000 lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew, luggage and fuel--Falcon Heavy can lift more than twice the payload of the next closest operational vehicle, the Delta IV Heavy, at one-third the cost.',
        reserved: false,
      },
    ],
    missions: [
      {
        id: 1,
        mission: 'Iridium NEXT',
        description: 'In 2017, Iridium began launching Iridium NEXT, a second-generation worldwide network of telecommunications satellites, consisting of 66 active satellites, with another nine in-orbit spares and six on-ground spares. These satellites will incorporate features such as data transmission that were not emphasized in the original design. The constellation will provide L-band data speeds of up to 128 kbit/s to mobile terminals, up to 1.5 Mbit/s to Iridium Pilot marine terminals, and high-speed Ka-band service of up to 8 Mbit/s to fixed/transportable terminals. The next-generation terminals and service are expected to be commercially available by the end of 2018. However, Iridiums proposed use of its next-generation satellites has raised concerns the service will harmfully interfere with GPS devices. The satellites will incorporate a secondary payload for Aireon, a space-qualified ADS-B data receiver. This is for use by air traffic control and, via FlightAware, for use by airlines. A tertiary payload on 58 satellites is a marine AIS ship-tracker receiver, for Canadian company exactEarth Ltd. Iridium can also be used to provide a data link to other satellites in space, enabling command and control of other space assets regardless of the position of ground stations and gateways.',
        reserved: false,
      },
    ],
  };

  return configureStore({
    preloadedState,
    reducer: {
      rockets,
      missions,
    },
  });
};

describe('App snapshots', () => {
  test('It should render the app', () => {
    const store = createStore();
    const app = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>,
    );

    expect(app).toMatchSnapshot();
  });

  test('It should render the Rockets page', () => {
    const store = createStore();
    const rocketsPage = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(rocketsPage).toMatchSnapshot();
  });
});
