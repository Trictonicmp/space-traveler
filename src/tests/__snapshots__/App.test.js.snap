// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshots It should render the Rockets page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="rocketsContainer"
      >
        <li
          class="rocket"
        >
          <img
            alt="Falcon Heavy"
            class="rocketImg"
          />
          <div
            class="rocketDetails"
          >
            <h2
              class="rocketTitle"
            >
              Falcon Heavy
            </h2>
            <p>
              With the ability to lift into orbit over 54 metric tons (119,000 lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew, luggage and fuel--Falcon Heavy can lift more than twice the payload of the next closest operational vehicle, the Delta IV Heavy, at one-third the cost.
            </p>
            <button
              class="
            button
            blue
            reserveButton
          "
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="rocketsContainer"
    >
      <li
        class="rocket"
      >
        <img
          alt="Falcon Heavy"
          class="rocketImg"
        />
        <div
          class="rocketDetails"
        >
          <h2
            class="rocketTitle"
          >
            Falcon Heavy
          </h2>
          <p>
            With the ability to lift into orbit over 54 metric tons (119,000 lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew, luggage and fuel--Falcon Heavy can lift more than twice the payload of the next closest operational vehicle, the Delta IV Heavy, at one-third the cost.
          </p>
          <button
            class="
            button
            blue
            reserveButton
          "
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App snapshots It should render the app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav"
      >
        <div
          class="brand d-flex"
        >
          <img
            alt="Logo"
            src="logo.png"
            width="70px"
          />
          <h2>
            Space Travelers’ Hub
          </h2>
        </div>
        <ul
          class="nav-links"
        >
          <a
            href="/"
          >
            Rockets
          </a>
          <a
            href="/mission"
          >
            Missions
          </a>
          <a
            class="profile-link"
            href="/profile"
          >
            Profile
          </a>
        </ul>
      </nav>
      <ul
        class="rocketsContainer"
      >
        <li
          class="rocket"
        >
          <img
            alt="Falcon Heavy"
            class="rocketImg"
          />
          <div
            class="rocketDetails"
          >
            <h2
              class="rocketTitle"
            >
              Falcon Heavy
            </h2>
            <p>
              With the ability to lift into orbit over 54 metric tons (119,000 lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew, luggage and fuel--Falcon Heavy can lift more than twice the payload of the next closest operational vehicle, the Delta IV Heavy, at one-third the cost.
            </p>
            <button
              class="
            button
            blue
            reserveButton
          "
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav"
    >
      <div
        class="brand d-flex"
      >
        <img
          alt="Logo"
          src="logo.png"
          width="70px"
        />
        <h2>
          Space Travelers’ Hub
        </h2>
      </div>
      <ul
        class="nav-links"
      >
        <a
          href="/"
        >
          Rockets
        </a>
        <a
          href="/mission"
        >
          Missions
        </a>
        <a
          class="profile-link"
          href="/profile"
        >
          Profile
        </a>
      </ul>
    </nav>
    <ul
      class="rocketsContainer"
    >
      <li
        class="rocket"
      >
        <img
          alt="Falcon Heavy"
          class="rocketImg"
        />
        <div
          class="rocketDetails"
        >
          <h2
            class="rocketTitle"
          >
            Falcon Heavy
          </h2>
          <p>
            With the ability to lift into orbit over 54 metric tons (119,000 lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew, luggage and fuel--Falcon Heavy can lift more than twice the payload of the next closest operational vehicle, the Delta IV Heavy, at one-third the cost.
          </p>
          <button
            class="
            button
            blue
            reserveButton
          "
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
